### PR TITLE
Added option to test_simple.py to directly predict depth. Plus comments in dataloader

### PR DESCRIPTION
**Pull request by mdfirman**
* Added comments to dataloader to help with custom intrinsics 
* Added flag `--pred_depth` to `test_simple.py` to make it easier to understand how to directly estimate depth.

### CODE REVIEW

**Comment by mdfirman** [comment]
Runs without errors. Predictions on our teaser images are as so:

image
image

File: test_simple.py [hunk 1]
```diff
@@ -46,6 +47,10 @@ def parse_args():
     parser.add_argument("--no_cuda",
                         help='if set, disables CUDA',
                         action='store_true')
+    parser.add_argument("--pred_depth",
```
Comment (mrharicot) [inline test_simple.py:50]: `--pred_metric_depth`

File: test_simple.py [hunk 2]
```diff
@@ -129,9 +138,13 @@ def test_simple(args):
 
             # Saving numpy file
             output_name = os.path.splitext(os.path.basename(image_path))[0]
-            name_dest_npy = os.path.join(output_directory, "{}_disp.npy".format(output_name))
-            scaled_disp, _ = disp_to_depth(disp, 0.1, 100)
-            np.save(name_dest_npy, scaled_disp.cpu().numpy())
+            scaled_disp, depth = disp_to_depth(disp, 0.1, 100)
+            if args.pred_depth:
+                name_dest_npy = os.path.join(output_directory, "{}_depth.npy".format(output_name))
+                np.save(name_dest_npy, STEREO_SCALE_FACTOR * depth.cpu().numpy())
```
Comment (mrharicot) [inline test_simple.py:144]: To bring the point home we should add:
```python
name_dest_npy = os.path.join(output_directory, "{}_metric_depth.npy".format(output_name))
metric_depth = STEREO_SCALE_FACTOR * depth
np.save(name_dest_npy, metric_depth.cpu().numpy())
```

File: test_simple.py [hunk 3]
```diff
@@ -46,6 +47,10 @@ def parse_args():
     parser.add_argument("--no_cuda",
                         help='if set, disables CUDA',
                         action='store_true')
+    parser.add_argument("--pred_depth",
+                        help='if set, predicts metric depth instead of disparity. (This only '
+                             'makes sense for stereo-trained models).',
```
Comment (mrharicot) [inline test_simple.py:52]: Should we also specify that this is only for KITTI?

**Review by mrharicot — APPROVED** [review]
Looks great, thanks for adding these!
I just have a few comments on the wording and the option name to add metric to bring the point home.

File: test_simple.py (hunk 1 above)
Comment (mdfirman) [reply test_simple.py:50]: Good idea, done

File: test_simple.py (hunk 3 above)
Comment (mdfirman) [reply test_simple.py:52]: Good idea, done

File: test_simple.py (hunk 2 above)
Comment (mdfirman) [reply test_simple.py:144]: Good idea, done

**Pull request merged**